test/mount/extensions/autostart_test.clj

```diff
@@ -0,0 +1,34 @@
+(ns mount.extensions.autostart-test
+  (:require [clojure.test :refer [deftest is testing]]
+            [mount.extensions.autostart :as sut]
+            [mount.extensions.explicit-deps :as deps]
+            [mount.lite :as mount]))
+
+(sut/defstate first
+  :dependencies []
+  :start 42)
+
+(sut/defstate second
+  :dependencies []
+  :start 420)
+
+(sut/defstate third
+  :dependencies [#'first #'second]
+  :start (+ @first @second))
+
+(defn fully?
+  [state]
+  (= {#'first state #'second state #'third state} (mount/status)))
+
+(deftest autostart-test
+  (testing "with default autostart-fn"
+    (= 462 @third)
+    (is (fully? :started))
+    (mount/stop)
+    (is (fully? :stopped)))
+  (testing "with custom autostart-fn"
+    (sut/set-autostart-fn! deps/start)
+    (= 462 @third)
+    (is (fully? :started))
+    (mount/stop)
+    (is (fully? :stopped))))
```

Comment:
I think the autostart-fn should be reset, in case the tests are run a second time within the same REPL session for example.
```suggestion
    (try
      (sut/set-autostart-fn! deps/start)
      (= 462 @third)
      (is (fully? :started))
      (mount/stop)
      (is (fully? :stopped))
      (finally
        (sut/set-autostart-fn! mount/start)))))
```